frontend/layouts/discover-page/helpers.ts

```diff
@@ -47,3 +58,123 @@ export const useSortingByOptions = (target?: SortingByTargetType): SortingOption
     ...(sortingOptions[target] || []),
   ]) as SortingOptionType[];
 };
+
+const pathFilters = {
+  [Paths.Projects]: [
+    'filter[category]',
+    'filter[sdg]',
+    'filter[instrument_type]',
+    'filter[ticket_size]',
+    'filter[impact]',
+    'filter[only_verified]',
+    'filter[full_text]',
+    'filter[priority_landscape]',
+  ],
+  [Paths.ProjectDevelopers]: [
+    'filter[category]',
+    'filter[impact]',
+    'filter[full_text]',
+    'filter[priority_landscape]',
+  ],
+  [Paths.Investors]: [
+    'filter[category]',
+    'filter[sdg]',
+    'filter[instrument_type]',
+    'filter[ticket_size]',
+    'filter[impact]',
+    'filter[full_text]',
+  ],
+  [Paths.OpenCalls]: [
+    'filter[sdg]',
+    'filter[instrument_type]',
+    'filter[ticket_size]',
+    'filter[only_verified]',
+    'filter[full_text]',
+  ],
+};
+
+const getPathParams = (
+  path: string,
+  pathname: string,
+  queryParams: { [key: string]: string | string[] }
+) => {
+  const { page, sorting, search, ...filters } = queryParams;
+
+  // Get only applicable filters for each path to avoid unnecessary fetches
+  const pathFilter = pickBy(filters, (_, key) => pathFilters[path].includes(key));
```

Comment:
I'm not really a fan of the list of filters which apply to each tab above and that is used by this line.

I understand that with it, when the user applies a filter that is not available for a specific tab, we avoid a request to fetch again the number of results for this tab.

That said, we're making the front-end code aware of which filters exists and for which tab. Previously, only the back-end contained that information.This means that if tomorrow the BE enables an existing filter for tab that had no use for it, we will have to update the FE as well.

In my opinion, it would be better to make the FE as dumb as possible so as to simplify the maintenance of the code.

Please let me know if you disagree. I might have forgotten other cases in which this piece of code saves requests.

Reply:
I agree with you. The requests avoided here are not delaying the render of the main content, so we can keep them for the sake of maintenance the FE code.